packages/library-base/src/commonMain/kotlin/io/realm/schema/RealmSchema.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.schema
+
+import kotlin.reflect.KType
+
+interface RealmSchema {
+    // Alternatively as Map<String, RealmClass>
+    val classes: Set<RealmClass>
+}
+
+interface RealmClass {
+    val name: String
+    val primaryKey : RealmProperty?
+    val embedded: Boolean
+    // Alternatively as Map<String, RealmProperty>
+    val properties: Set<RealmProperty>
+}
+
+interface RealmProperty {
+    val name: String // val publicName: String or should we just support mappings already?
+    val index: Boolean
+    val primaryKey: Boolean
+    val type: RealmPropertyType
```

Comment:
Not sure this is enough to also cover collection types? I think we talked about using the same bitfield approach that Core uses, but maybe there are other ways of doing it 🤔